### PR TITLE
Added authorization expired response tests

### PR DESCRIPTION
**Pull request by injectives**
These tests verify that the driver handles the authorization expired response correctly. Such response can be returned by the server when authorization info that is maintained on the server side expires.

### CODE REVIEW

**Comment by injectives** [comment]
@gjmwoods please check that the scripts match the server-side behaviour.

**Comment by injectives** [comment]
The Java driver is expected to fail until we have merged this: https://github.com/neo4j/neo4j-java-driver/pull/886

**Review by gjmwoods — APPROVED** [review]
looks like a very comprehensive suite of tests!

File: tests/stub/authorization.py [hunk 1]
```diff
@@ -0,0 +1,634 @@
...
+
+def get_extra_hello_props():
+    if get_driver_name() in ["java"]:
+        return ', "realm": ""'
+    elif get_driver_name() in ["javascript"]:
+        return ', "realm": "", "ticket": ""'
+    return ""
+
+
+class Tests(TestkitTestCase):
```
Comment (robsdedude) [inline tests/stub/authorization.py]: Maybe give it a more descriptive name like `AuthorizationTests` or such.

Comment (injectives) [reply tests/stub/authorization.py]: Good point, I have updated the PR, thank you 👍

File: tests/stub/routing.py [hunk 2]
```diff
@@ -129,8 +129,8 @@ def router_with_bookmarks_script_create_adb(self):
         C: HELLO {"scheme": "basic", "credentials": "c", "principal": "p", "user_agent": "007", "routing": #HELLO_ROUTINGCTX# #EXTRA_HELLO_PROPS#}
         S: SUCCESS {"server": "#SERVER_AGENT#", "connection_id": "bolt-123456789"}
         C: BEGIN {"db": "system"}
-        C: RUN "CREATE database foo" {} {}
         S: SUCCESS {}
+        C: RUN "CREATE database foo" {} {}
```
Comment (robsdedude) [inline tests/stub/routing.py:133]: I think I've seen many of those, where the stub server expects the driver to pipeline multiple requests. We should probably (in another PR) change that.

**Review by robsdedude — APPROVED** [review]
I haven't looked in great detail at every singe test, but overall they look good to me. If there are minor problems, I'll probably discover them while adjusting the Python driver.

**Pull request merged**